typing/typetexp.ml

```diff
@@ -380,6 +380,46 @@ let transl_type_param env styp =
   Builtin_attributes.warning_scope styp.ptyp_attributes
     (fun () -> transl_type_param env styp)
 
+
+let new_pre_univar ?name () =
+  let v = newvar ?name () in pre_univars := v :: !pre_univars; v
+
+type poly_univars = (string * type_expr) list
+let make_poly_univars vars =
+  List.map (fun name -> name, newvar ~name ()) vars
+
+let check_poly_univars env loc vars =
+  vars |> List.iter (fun (_, v) -> generalize v);
+  vars |> List.map (fun (name, ty1) ->
+    let v = Btype.proxy ty1 in
+    begin match get_desc v with
+    | Tvar name when get_level v = Btype.generic_level ->
+       set_type_desc v (Tunivar name)
+    | _ ->
+       raise (Error (loc, env, Cannot_quantify(name, v)))
+    end;
+    v)
+
+let instance_poly_univars env loc vars =
+  let vs = check_poly_univars env loc vars in
+  vs |> List.iter (fun v ->
+    match get_desc v with
+    | Tunivar name ->
+       set_type_desc v (Tvar name)
+    | _ -> assert false);
+  vs
+
+
+type policy = Fixed | Extensible | Univars
```

Comment:
The stuff above looks like an erroneous resolution of a merge conflict with the recent TyVarEnv refactoring. i would remove all of it.

Reply:
Yes, this was fixed in commit 2bc28b40852a173e57e08a567e9646d77d5bd4e9. I can fix it, but I'm more than happy for the commits to be squashed when merging